index.js

```diff
@@ -397,6 +410,46 @@ async function fastifyStatic (fastify, opts) {
     const routeConfig = req.routeOptions?.config || req.routeConfig
     pumpSendToReply(req, reply, routeConfig.file, routeConfig.rootPath)
   }
+
+  async function generateHashForFiles () {
+    fileHashes = new Map()
+
+    const roots = Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]
+    for (let rootPath of roots) {
+      rootPath = rootPath.split(path.win32.sep).join(path.posix.sep)
+      if (!rootPath.endsWith('/')) {
+        rootPath += '/'
+      }
+
+      const files = await glob(`${rootPath}**/**`, { follow: true, nodir: true, dot: opts.serveDotFiles })
```

Comment:
wonder if this part of the code would benefit from a promise pool, so that it isn't processed one-by-one. or realistically filesystem operations don't benefit from parallelization?

Reply:
Yeah when it's not a network operation, I really don't know which is better

Maybe I could find a benchmark somewhere 😅